pkg/lvm/lvm_util.go

```diff
@@ -518,6 +523,17 @@ func lvThinExists(vg string, name string) bool {
 	return name == strings.TrimSpace(string(out))
 }
 
+// snapshotExists checks if a snapshot volume exists given the name of the volume.
+func isSnapshotExists(vg, snapVolumeName string) (bool, error) {
+	cmd := exec.Command("lvs", vg+"/"+snapVolumeName, "--noheadings", "-o", "lv_name")
```

Comment:
This change is required , since thin-snapshot can not be found in `/dev/vol-group/snapshot-id` path using `os.State()`.